src/common/errfun.F

```diff
@@ -0,0 +1,485 @@
+
+! Copyright (c) 2016 Anton Shterenlikht, The University of Bristol, UK
+!
+! Redistribution and use in source and binary forms, with or without
+! modification, are permitted provided that the following conditions are
+! met:
+!
+! 1. Redistributions of source code must retain the above copyright
+! notice, this list of conditions and the following disclaimer.
+!
+! 2. Redistributions in binary form must reproduce the above copyright
+! notice, this list of conditions and the following disclaimer in the
+! documentation and/or other materials provided with the distribution.
+!
+! THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+! IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+! TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+! PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+! HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+! SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+! TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+! PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+! LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+! NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+! SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+!
+! Module with error function and related functions.
+! Two algorithms are implemented: Poppe and Wijers (wpop, erf_pop)
+! and Zaghloul and Ali (wzag, erf_zag). The second algorithm is
+! supposed to be more accurate for some values. However, the first
+! algorithm can be over an order of magnitude faster.
+!
+! This code was written before the author became aware of
+! M. R. Zaghloul, Remark on "Algorithm 916: Computing the
+! Faddeyeva and Voight functions": efficiency improvements and
+! Fortran translation, ACM Trans. Math. Software 42, Article 26, 2016.
+
+! **************************************************************************************************
+!> \brief Module to compute the error function of a complex argument
+!> \par History
+!>      08.2025 Adapted to use CP2K intrinsics and constants
+!> \author Stefano Battaglia
+! **************************************************************************************************
+MODULE errfun
+
+   USE kinds,                           ONLY: dp
+   USE mathconstants,                   ONLY: pi
+
+   IMPLICIT NONE
+
+   REAL(kind=dp), PARAMETER :: zero = 0.0_dp, one = 1.0_dp, &
+                               two = 2.0_dp, half = 0.5_dp, rmin = TINY(one), &
+                               eps0 = EPSILON(one), sqrt_log_rmin = SQRT(-LOG(rmin)), &
+                               pi2 = pi*pi, one_sqrt_pi = one/SQRT(pi)
+   COMPLEX(kind=dp), PARAMETER :: &
+      cmplxj = CMPLX(zero, one, kind=dp), &
+      cmplx0 = CMPLX(zero, zero, kind=dp)
+
+   PRIVATE
+   PUBLIC :: wpop, wzag, erf_pop, erf_zag
```

Comment:
Please choose better names than w or erf which is either too generic or a bit misleading for other developers. I would also rather vote for something like faddeyeva_fast and faddeyeva_accurate to build in the difference between both implementations.

Reply:
Sure, I'll update that now.